src/commands/edge_app.rs

```diff
@@ -95,6 +95,54 @@ impl EdgeAppCommand {
         Ok(())
     }
 
+    pub fn init(&self, name: &str, path: &Path) -> Result<(), CommandError> {
+        let parent_dir_path = path.parent().ok_or(CommandError::FileSystemError(
+            "Can not obtain edge app root directory.".to_owned(),
+        ))?;
+        let index_html_path = parent_dir_path.join("index.html");
+
+        if !(Path::new(&path).exists() && Path::new(&index_html_path).exists()) {
+            return Err(CommandError::FileSystemError(format!(
+                "The directory {} should contain screenly.yml and index.html files",
+                parent_dir_path.display()
+            )));
+        }
+
+        let data = fs::read_to_string(path)?;
+        let mut manifest: EdgeAppManifest = serde_yaml::from_str(&data)?;
+
+        if manifest.app_id != "" {
```

Comment:
Nope app_id can't be empty. We are going to not allow that. Either it's not present there or non empty. Though I guess at the moment it works

Reply:
You can change it now to be an Option<String> in manifest though leaving non empty part. If option is None your code should work